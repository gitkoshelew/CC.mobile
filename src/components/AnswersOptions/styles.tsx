import styled from 'styled-components/native';
import {StyleSheet} from 'react-native';
import {DefaultTheme} from 'styled-components';

export const AnswerRadioContainer = styled.View`
  flex-direction: row;
  align-items: center;
  padding: 14px 16px;
  width: 300px;
  background-color: ${props => props.theme.layout};
  margin-bottom: 20px;
  border: 1px solid ${props => props.theme.border};
  border-radius: 15px;
`;

export const ViewMarginRight = styled.View`
  margin-right: 18px;
`;

export const styles = StyleSheet.create((theme?: DefaultTheme) => ({
  textRadio: {
    color: theme?.textMainColor,
    fontSize: 16,
    fontFamily: 'Montserrat-Regular',
  },
}));
