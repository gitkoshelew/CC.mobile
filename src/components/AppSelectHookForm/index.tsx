import {Control, Controller, FieldPath, FieldValues} from 'react-hook-form';
import {TextInputProps} from 'react-native';
import React from 'react';
import {AppSelect} from '@src/components/ui/AppSelect/index';
import {TextBox, BlockBox} from '@src/components/ui/ReadyStyles/Boxes/index';

type TextInputHookFormPropsType<T extends FieldValues> = TextInputProps & {
  name: FieldPath<T>;
  data: string[];
  size: 's' | 'm';
  type: 'primary' | 'secondary';
  label?: string | null;
  rules?: Object;
  control: Control<T>;
  onAppSelect?: (value: string, onPress: (value: string) => void) => void;
};

export const AppSelectHookForm = <T extends FieldValues>({
  data,
  name,
  size,
  type,
  label,
  rules = {},
  control,
  onAppSelect,
}: TextInputHookFormPropsType<T>) => {
  const handlerSelect = (value: string, onPress: (value: string) => void) => {
    onAppSelect && onAppSelect(value, onPress);
  };

  return (
    <BlockBox>
      {label && <TextBox>{label}</TextBox>}
      <Controller
        control={control}
        name={name}
        rules={rules}
        render={({field: {onChange, value}}) => (
          <AppSelect
            value={value}
            size={size}
            data={data}
            type={type}
            onSelect={(selected: string) => handlerSelect(selected, onChange)}
          />
        )}
      />
    </BlockBox>
  );
};
