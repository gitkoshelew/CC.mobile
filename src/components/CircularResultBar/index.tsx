import CircularProgress from 'react-native-circular-progress-indicator';
import {Color} from '@theme/colors';
import {StyleSheet} from 'react-native';
import {ViewCenter} from '../ui/ReadyStyles/Containers';
import {useContext} from 'react';
import {ThemeContext} from 'styled-components/native';

interface IProps {
  result: number;
}

export const CircularResultBar = ({result}: IProps) => {
  const theme = useContext(ThemeContext);

  return (
    <ViewCenter>
      <CircularProgress
        value={result}
        radius={83}
        progressValueStyle={styles.progressValueStyle}
        valueSuffix={'%'}
        duration={1500}
        activeStrokeWidth={26}
        activeStrokeColor={Color.BlueLight}
        inActiveStrokeWidth={26}
        inActiveStrokeColor={theme.box}
      />
    </ViewCenter>
  );
};

const styles = StyleSheet.create({
  progressValueStyle: {
    fontSize: 38,
    fontFamily: 'Montserrat-Regular',
    color: Color.BlueLight,
  },
});
