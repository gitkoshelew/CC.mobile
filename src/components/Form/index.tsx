import {FormSignIn} from '@src/components/Form/FormSignIn';
import {FormSignUp} from '@src/components/Form/FormSignUp';

export type FormType = {
  isLogForm?: boolean;
  setIsLogForm: (value: boolean) => void;
};

export const Form = ({isLogForm, setIsLogForm}: FormType) => {
  return isLogForm ? (
    <FormSignIn setIsLogForm={setIsLogForm} />
  ) : (
    <FormSignUp setIsLogForm={setIsLogForm} />
  );
};
