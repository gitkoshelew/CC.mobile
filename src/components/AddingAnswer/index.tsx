import {TouchableOpacity, View} from 'react-native';
import {styles} from './styles';
import {BlockAnswerBox} from '../ui/ReadyStyles/Boxes';
import {CustomTextInput} from '../ui/CustomTextInput';
import {CheckBox} from '../ui/CheckBox';
import AntDesign from 'react-native-vector-icons/AntDesign';
import {Color} from '@theme/colors';
import {useCallback, useEffect, useState} from 'react';
import {Control, Controller, useWatch} from 'react-hook-form';
import {useTranslation} from 'react-i18next';
import {CreateQuestionFieldType} from '@src/screens/CreateQuiz/components/CreateQuestion/CreateQuestion';

type AddingAnswerPropsType = {
  type: string;
  index: number;
  option: string;
  control: Control<CreateQuestionFieldType>;
  correctAnswer: string[];
  onPressDelete: (index: number) => void;
  isCheckingDuplicate: boolean;
  isDisabledDeleteBtn: boolean;
  onPressCorrectAnswer: (index: number, checked: boolean, textOption: string) => void;
};

export const AddingAnswer = (props: AddingAnswerPropsType) => {
  const {
    type,
    index,
    option,
    control,
    onPressDelete,
    correctAnswer,
    isCheckingDuplicate,
    isDisabledDeleteBtn,
    onPressCorrectAnswer,
  } = props;
  const {t} = useTranslation(['validationFields']);
  const findCorrectInput = correctAnswer.find(el => el === option);
  const [isChecked, setIsChecked] = useState(false);
  const [correctInput, setCorrectInput] = useState(findCorrectInput);
  const isCurrentOptionText = useWatch({
    name: `options.${index}.option`,
    control,
  });

  const onPressCorrectAnswerHandler = useCallback(
    (checked: boolean) => {
      setIsChecked(checked);
      setCorrectInput(isCurrentOptionText);
      onPressCorrectAnswer(index, checked, isCurrentOptionText);
    },
    [index, isCurrentOptionText, onPressCorrectAnswer],
  );
  const onPressDeletePressed = useCallback(() => {
    onPressDelete(index);
  }, [index, onPressDelete]);

  useEffect(() => {
    if (correctInput !== isCurrentOptionText) {
      onPressCorrectAnswer(index, false, correctInput!);
      setIsChecked(false);
      setCorrectInput('');
    }
  }, [index, correctInput, isCurrentOptionText, onPressCorrectAnswer]);

  useEffect(() => {
    setIsChecked(correctAnswer.includes(isCurrentOptionText));
  }, [isCurrentOptionText, correctAnswer, option]);

  const isSingle = type === 'single';
  const isSingleDisabled = (!isChecked && correctAnswer.length) || !isCurrentOptionText;
  const isMultipleDisabled = !isCurrentOptionText;

  const isDisabledDeleteOption = isDisabledDeleteBtn || isChecked;

  return (
    <BlockAnswerBox>
      <View style={styles.inputBox}>
        <Controller
          render={({field: {onChange, value, onBlur}, fieldState: {error}}) => (
            <CustomTextInput
              onChangeText={onChange}
              value={value}
              onBlur={onBlur}
              error={error}
            />
          )}
          rules={{
            required: `${t('option.required')}`,
            maxLength: {
              value: 50,
              message: `${t('validationFields.option.minLength')}`,
            },
          }}
          name={`options.${index}.option`}
          control={props.control}
        />
      </View>
      <View style={styles.picker}>
        <CheckBox
          onPress={onPressCorrectAnswerHandler}
          isChecked={isChecked}
          disabled={Boolean(isSingle ? isSingleDisabled : isMultipleDisabled)}
          fillColor={isCheckingDuplicate ? Color.Red : Color.GreenLight}
        />
        <TouchableOpacity
          style={isDisabledDeleteOption && styles.disabled}
          onPress={onPressDeletePressed}
          disabled={isDisabledDeleteOption}>
          <AntDesign name="minuscircleo" size={30} color={Color.Red} />
        </TouchableOpacity>
      </View>
    </BlockAnswerBox>
  );
};
