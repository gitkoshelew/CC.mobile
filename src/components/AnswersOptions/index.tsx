import {useContext} from 'react';
import {View} from 'react-native';
import RadioForm, {RadioButtonInput, RadioButtonLabel} from 'react-native-simple-radio-button';
import {AnswerRadioContainer, styles, ViewMarginRight} from './styles';
import {Color} from '@theme/colors';
import {MultipleCheckboxes} from '@src/components/MultipleCheckboxes/MultipleCheckboxes';
import {CustomText} from '@src/components/ui/ReadyStyles/Boxes/index';
import {ThemeContext} from 'styled-components/native';

type AnswersOptionsPropsType = {
  onPress: (label: string, value: number) => void;
  data: string[];
  answerType: string;
  selected?: number;
  dataOptions: IDataOptions[];
  onPressCheck: (label: string, value: number, isChecked: boolean) => void;
};
export type IDataOptions = {
  label: string;
  value: number;
  isChecked: boolean;
};

export const AnswersOptions = ({
  onPress,
  onPressCheck,
  dataOptions,
  data,
  selected,
  answerType,
}: AnswersOptionsPropsType) => {
  const theme = useContext(ThemeContext);

  const onPressRadio = (value: number) => {
    let answer: string = data[value];
    onPress(answer, value);
  };
  const handlerPressCheck = (label: string, value: number, isChecked: boolean) => {
    onPressCheck(label, value, isChecked);
  };

  return (
    <View>
      <RadioForm formHorizontal={false} animation={true}>
        {dataOptions.map((obj, i) => (
          <AnswerRadioContainer key={i}>
            <ViewMarginRight>
              {answerType === 'single' ? (
                <RadioButtonInput
                  obj={obj}
                  index={i}
                  isSelected={selected === i}
                  onPress={onPressRadio}
                  buttonOuterColor={selected === i ? Color.BlueLight : Color.GrayStrongDark}
                  buttonSize={15}
                  buttonOuterSize={30}
                />
              ) : (
                <MultipleCheckboxes item={obj} onPress={handlerPressCheck} />
              )}
            </ViewMarginRight>
            {answerType === 'single' ? (
              <RadioButtonLabel
                obj={obj}
                index={i}
                labelHorizontal={false}
                onPress={onPressRadio}
                labelStyle={styles(theme).textRadio}
              />
            ) : (
              <CustomText key={i} fs="16px">
                {obj.label}
              </CustomText>
            )}
          </AnswerRadioContainer>
        ))}
      </RadioForm>
    </View>
  );
};
