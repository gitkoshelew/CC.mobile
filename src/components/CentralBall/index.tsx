import React from 'react';
import {Color} from '@theme/colors';
import {Text} from 'react-native';
import LinearGradient from 'react-native-linear-gradient';
import {styles} from '@src/components/CentralBall/styles';
import {ViewDynamicFlex} from '@src/components/ui/ReadyStyles/Containers/index';

type CentralBallPropsType = {
  value: number;
  description: string;
};

export const CentralBall = ({value, description}: CentralBallPropsType) => {
  return (
    <LinearGradient
      start={{x: 0.0, y: 0.05}}
      end={{x: 0.6, y: 1.0}}
      colors={[Color.BlueLight, Color.DarkBlue]}
      style={styles.container}>
      <ViewDynamicFlex alignI="center" justifyC="center">
        <Text style={styles.value}>{value}</Text>
        <Text style={styles.description}>{description}</Text>
      </ViewDynamicFlex>
    </LinearGradient>
  );
};
