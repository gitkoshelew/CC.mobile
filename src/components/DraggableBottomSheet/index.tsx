import {useEffect, useRef, useState} from 'react';
import {
  Animated,
  Dimensions,
  ImageBackground,
  PanResponder,
  ScaledSize,
  View,
} from 'react-native';
import {Content} from '@src/components/DraggableBottomSheet/Content/Content';
import {useAppDispatch, useAppSelector} from '@hooks/hooks';
import {getStyles} from '@src/components/DraggableBottomSheet/styles';
import {setIsScrollEnabled} from '@src/bll/appReducer';

export const {width: WINDOW_WIDTH, height: WINDOW_HEIGHT}: ScaledSize =
  Dimensions.get('window');

const BOTTOM_SHEET_MIN_HEIGHT = WINDOW_HEIGHT * 0.03;
const MAX_DOWNWARD_TRANSLATE_Y = 0;
const DRAG_THRESHOLD = 30;

export const DraggableBottomSheet = () => {
  const dispatch = useAppDispatch();
  const [isOpen, setIsOpen] = useState(false);
  const [isLogForm, setIsLogForm] = useState(false);

  const isLoggedIn = useAppSelector(state => state.authReducer.isLoggedIn);

  const BOTTOM_SHEET_MAX_HEIGHT = isLoggedIn
    ? WINDOW_HEIGHT * 0.26
    : isLogForm
    ? WINDOW_HEIGHT * 0.5
    : WINDOW_HEIGHT * 0.68;
  const MAX_UPWARD_TRANSLATE_Y = BOTTOM_SHEET_MIN_HEIGHT - BOTTOM_SHEET_MAX_HEIGHT;

  const animatedValue = useRef(new Animated.Value(0)).current;
  const lastGestureDy = useRef(0);
  const panResponder = useRef(
    PanResponder.create({
      onStartShouldSetPanResponder: () => true,
      onPanResponderGrant: () => {
        dispatch(setIsScrollEnabled(false));
        animatedValue.setOffset(lastGestureDy.current);
      },
      onPanResponderEnd: () => {
        dispatch(setIsScrollEnabled(true));
      },
      onPanResponderMove: (e, gesture) => {
        animatedValue.setValue(gesture.dy);
      },
      onPanResponderRelease: (e, gesture) => {
        animatedValue.flattenOffset();
        if (gesture.dy > 0) {
          if (gesture.dy <= DRAG_THRESHOLD) {
            springAnimation('up');
          } else {
            springAnimation('down');
          }
        } else {
          if (gesture.dy >= -DRAG_THRESHOLD) {
            springAnimation('down');
          } else {
            springAnimation('up');
          }
        }
      },
    }),
  ).current;

  const springAnimation = (direction: 'up' | 'down') => {
    lastGestureDy.current =
      direction === 'down' ? MAX_DOWNWARD_TRANSLATE_Y : MAX_UPWARD_TRANSLATE_Y;
    Animated.spring(animatedValue, {
      toValue: lastGestureDy.current,
      useNativeDriver: true,
    }).start();
    lastGestureDy.current >= 0 ? setIsOpen(false) : setIsOpen(true);
  };

  const bottomSheetAnimation = {
    transform: [
      {
        translateY: animatedValue.interpolate({
          inputRange: [MAX_UPWARD_TRANSLATE_Y, MAX_DOWNWARD_TRANSLATE_Y],
          outputRange: [MAX_UPWARD_TRANSLATE_Y, MAX_DOWNWARD_TRANSLATE_Y],
          extrapolate: 'clamp',
        }),
      },
    ],
  };

  useEffect(() => {
    !isLoggedIn && setIsLogForm(true);
  }, [isLoggedIn]);

  return (
    <View style={getStyles().container}>
      <Animated.View
        style={[
          getStyles(BOTTOM_SHEET_MAX_HEIGHT, MAX_UPWARD_TRANSLATE_Y).bottomSheet,
          bottomSheetAnimation,
        ]}>
        <ImageBackground
          style={getStyles().imageBackground}
          source={require('../../assets/images/background-second.png')}
          resizeMode="stretch">
          <View style={getStyles().draggableArea} {...panResponder.panHandlers}>
            <View style={getStyles().dragHandle} />
          </View>
          <Content isOpen={isOpen} isLogForm={isLogForm} setIsLogForm={setIsLogForm} />
        </ImageBackground>
      </Animated.View>
    </View>
  );
};
