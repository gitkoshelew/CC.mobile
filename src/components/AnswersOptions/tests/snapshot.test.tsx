import {act, fireEvent, render, screen} from '@testing-library/react-native';
import {AnswersOptions} from '@src/components/AnswersOptions';
import {AnswersOptionsProps} from '@src/components/AnswersOptions/tests/functional.test';
import {Provider} from 'react-redux';
import store from '@src/bll/store/store';
import React from 'react';

test('Should render AnswersOptions and selected radio button', () => {
  const answersOptions = render(
    <Provider store={store}>
      render(
      <AnswersOptions {...AnswersOptionsProps} />
      );
    </Provider>,
  );

  expect(answersOptions).toMatchSnapshot();

  act(() => {
    fireEvent.press(screen.getByText('one'));
  });

  expect(answersOptions).toMatchSnapshot();
});
