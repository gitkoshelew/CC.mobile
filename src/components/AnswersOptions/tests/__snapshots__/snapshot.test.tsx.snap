// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render AnswersOptions and selected radio button 1`] = `
<View>
  <View
    style={
      [
        undefined,
        undefined,
        false,
      ]
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "#F0F0F0",
            "borderColor": "#CCD2E3",
            "borderRadius": 15,
            "borderStyle": "solid",
            "borderWidth": 1,
            "flexDirection": "row",
            "marginBottom": 20,
            "paddingBottom": 14,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 14,
            "width": 300,
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "marginRight": 18,
            },
          ]
        }
      >
        <View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "alignSelf": "center",
                "borderColor": "#49454F",
                "borderRadius": 12.5,
                "borderWidth": 3,
                "height": 25,
                "justifyContent": "center",
                "opacity": 1,
                "width": 25,
              }
            }
          >
            <View
              style={
                [
                  {
                    "borderRadius": 10,
                  },
                  false,
                  false,
                  false,
                ]
              }
            />
          </View>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            undefined,
            {
              "alignItems": "center",
              "alignSelf": "center",
              "flexDirection": "row",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "lineHeight": 20,
                "paddingLeft": 10,
              },
              {
                "paddingLeft": 0,
              },
              {
                "color": undefined,
              },
              {
                "fontFamily": "Montserrat-Regular",
                "fontSize": 16,
              },
            ]
          }
        >
          one
        </Text>
      </View>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "#F0F0F0",
            "borderColor": "#CCD2E3",
            "borderRadius": 15,
            "borderStyle": "solid",
            "borderWidth": 1,
            "flexDirection": "row",
            "marginBottom": 20,
            "paddingBottom": 14,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 14,
            "width": 300,
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "marginRight": 18,
            },
          ]
        }
      >
        <View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "alignSelf": "center",
                "borderColor": "#49454F",
                "borderRadius": 12.5,
                "borderWidth": 3,
                "height": 25,
                "justifyContent": "center",
                "opacity": 1,
                "width": 25,
              }
            }
          >
            <View
              style={
                [
                  {
                    "borderRadius": 10,
                  },
                  false,
                  false,
                  false,
                ]
              }
            />
          </View>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            undefined,
            {
              "alignItems": "center",
              "alignSelf": "center",
              "flexDirection": "row",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "lineHeight": 20,
                "paddingLeft": 10,
              },
              {
                "paddingLeft": 0,
              },
              {
                "color": undefined,
              },
              {
                "fontFamily": "Montserrat-Regular",
                "fontSize": 16,
              },
            ]
          }
        >
          two
        </Text>
      </View>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "#F0F0F0",
            "borderColor": "#CCD2E3",
            "borderRadius": 15,
            "borderStyle": "solid",
            "borderWidth": 1,
            "flexDirection": "row",
            "marginBottom": 20,
            "paddingBottom": 14,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 14,
            "width": 300,
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "marginRight": 18,
            },
          ]
        }
      >
        <View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "alignSelf": "center",
                "borderColor": "#49454F",
                "borderRadius": 12.5,
                "borderWidth": 3,
                "height": 25,
                "justifyContent": "center",
                "opacity": 1,
                "width": 25,
              }
            }
          >
            <View
              style={
                [
                  {
                    "borderRadius": 10,
                  },
                  false,
                  false,
                  false,
                ]
              }
            />
          </View>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            undefined,
            {
              "alignItems": "center",
              "alignSelf": "center",
              "flexDirection": "row",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "lineHeight": 20,
                "paddingLeft": 10,
              },
              {
                "paddingLeft": 0,
              },
              {
                "color": undefined,
              },
              {
                "fontFamily": "Montserrat-Regular",
                "fontSize": 16,
              },
            ]
          }
        >
          three
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Should render AnswersOptions and selected radio button 2`] = `
<View>
  <View
    style={
      [
        undefined,
        undefined,
        false,
      ]
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "#F0F0F0",
            "borderColor": "#CCD2E3",
            "borderRadius": 15,
            "borderStyle": "solid",
            "borderWidth": 1,
            "flexDirection": "row",
            "marginBottom": 20,
            "paddingBottom": 14,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 14,
            "width": 300,
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "marginRight": 18,
            },
          ]
        }
      >
        <View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "alignSelf": "center",
                "borderColor": "#6B9DE9",
                "borderRadius": 12.5,
                "borderWidth": 3,
                "height": 25,
                "justifyContent": "center",
                "opacity": 1,
                "width": 25,
              }
            }
          >
            <View
              style={
                [
                  {
                    "borderRadius": 10,
                  },
                  {
                    "backgroundColor": "#2196f3",
                    "height": 20,
                    "width": 20,
                  },
                  {
                    "borderRadius": 7.5,
                    "height": 15,
                    "width": 15,
                  },
                  {
                    "backgroundColor": "#2196f3",
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            undefined,
            {
              "alignItems": "center",
              "alignSelf": "center",
              "flexDirection": "row",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "lineHeight": 20,
                "paddingLeft": 10,
              },
              {
                "paddingLeft": 0,
              },
              {
                "color": undefined,
              },
              {
                "fontFamily": "Montserrat-Regular",
                "fontSize": 16,
              },
            ]
          }
        >
          one
        </Text>
      </View>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "#F0F0F0",
            "borderColor": "#CCD2E3",
            "borderRadius": 15,
            "borderStyle": "solid",
            "borderWidth": 1,
            "flexDirection": "row",
            "marginBottom": 20,
            "paddingBottom": 14,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 14,
            "width": 300,
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "marginRight": 18,
            },
          ]
        }
      >
        <View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "alignSelf": "center",
                "borderColor": "#49454F",
                "borderRadius": 12.5,
                "borderWidth": 3,
                "height": 25,
                "justifyContent": "center",
                "opacity": 1,
                "width": 25,
              }
            }
          >
            <View
              style={
                [
                  {
                    "borderRadius": 10,
                  },
                  false,
                  false,
                  false,
                ]
              }
            />
          </View>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            undefined,
            {
              "alignItems": "center",
              "alignSelf": "center",
              "flexDirection": "row",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "lineHeight": 20,
                "paddingLeft": 10,
              },
              {
                "paddingLeft": 0,
              },
              {
                "color": undefined,
              },
              {
                "fontFamily": "Montserrat-Regular",
                "fontSize": 16,
              },
            ]
          }
        >
          two
        </Text>
      </View>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "#F0F0F0",
            "borderColor": "#CCD2E3",
            "borderRadius": 15,
            "borderStyle": "solid",
            "borderWidth": 1,
            "flexDirection": "row",
            "marginBottom": 20,
            "paddingBottom": 14,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 14,
            "width": 300,
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "marginRight": 18,
            },
          ]
        }
      >
        <View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "alignSelf": "center",
                "borderColor": "#49454F",
                "borderRadius": 12.5,
                "borderWidth": 3,
                "height": 25,
                "justifyContent": "center",
                "opacity": 1,
                "width": 25,
              }
            }
          >
            <View
              style={
                [
                  {
                    "borderRadius": 10,
                  },
                  false,
                  false,
                  false,
                ]
              }
            />
          </View>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            undefined,
            {
              "alignItems": "center",
              "alignSelf": "center",
              "flexDirection": "row",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "lineHeight": 20,
                "paddingLeft": 10,
              },
              {
                "paddingLeft": 0,
              },
              {
                "color": undefined,
              },
              {
                "fontFamily": "Montserrat-Regular",
                "fontSize": 16,
              },
            ]
          }
        >
          three
        </Text>
      </View>
    </View>
  </View>
</View>
`;
